### PR TITLE
WIP: Building block on Exporting/Importing python environments with Dockerhub

### PR DESCRIPTION
**Pull request by DiSanchz**
This pull request contains the first version of the building block on how to import/export python environments with docker. In general is was doubting about splitting it in two separe building blocks (one for import another for export) but I would say that the result of a single one containing both processes is not too long. However this may need to be reconsidered if a substantial amount of new material is added. 

I also thought on referencing the g_cloud-docker building block during this block and vice-versa though I thought it may be better to first make any required fixes before going for that which in any case would just imply pretty minor modifications. 

May need a bit of polish here and there but for the rest, being it the first version, I would say ther are no other significant comments on my side. Let me know how you would improve, if you think the structure is appropriate for the content or if you detect any mistakes/inaccuracies so I can fix them!!

@shrabasteebanerjee 
@narendramukherjee

PD: for a mistake of mine in branching this PR also includes some of the modifications of the GPU addendum to the G-cloud/docker building block though this should be solved as soon as these (which are ready to merge) are merged. The relevant commits are only those from Jan 7th which affect the file "dockerhub.md".

### CODE REVIEW

**Comment by DiSanchz** [comment]
The last commits contain mostly various modifications, error fixes, and improvements in expression as suggested by Grammarly.

File: content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+After that the `RUN` instruction, which is used to execute commands within the container, is employed to install jupyter notebook as well as any packages that are necessary to replicate your environment. In this particular example, the "pytest" package is the only one that is going to be downloaded. Note that miniconda includes already some of the most widely used python packages such as Numpy and Pandas
+
+Lastly, the `RUN` instruction is employed again to create a new directory within the image's container that will be used to connect the container filesystem with your pc filesystem. Don't worry if you don't fully grasp this now, you will come back at this later in the building block. 
+
+Feel free to use this example dockerfile as a template to export your environment if it adjusts to your needs! For that, you should add to it the name of the packages included in your environment (instead of the included pytest) and save it as a text file without an extension named "dockerfile" (i.e. the file should be named "dockerfile" not "dockerfile.txt" or anything else, otherwise docker will not recognize it.)
```
Comment (narendramukherjee) [inline content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md:51]: Maybe we can also say that they don't have to manually write the names of the all the packages here - if they have an `environment.yml` file with all the packages listed (as is common with conda usage), they can also do something like `conda create <filename>`

File: content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+### Pushing your image to Docker Hub
+
+[Docker Hub](https://hub.docker.com/) is the most popular platform for hosting and distributing docker images, which are organized in repositories. One of the advantages of hosting your container image on Docker Hub is that anyone willing to use it (i.e. replicate your environment in this case) will be able to do so remotely simply by pulling from their command line.
+
+{{% tip %}}
+Before you try to push your image make sure you are properly logged in to your Docker Hub account. To do that run `docker login` in the command line and then introduce your Docker Hub username and password.
+{{% /tip %}}
+
+A Docker Hub repository is designed to contain one or multiple versions of a docker image, which are differentiated by their tag. It is important that you assign an appropriate tag to each of your images within a repository so you can appropriately differentiate them. To assign a tag to your recently built image you should type the following in the command line:
```
Comment (narendramukherjee) [inline content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md:99]: If you don't specify a tag it isn't a huge problem: docker/DockerHub will just assign the `latest` tag automatically to the latest build of the image

File: content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+</p>
+
+<p align = "center">
+<img src = "../img/docker-compose-your-env.png" width="500">
+<figcaption> Change the name of the directory where the volume is mounted to whichever you prefer. Just make sure that it is the same across both the dockerfile and docker-compose files. </figcaption>
+</p>
+
+### Creating new versions of your images with "docker commit"
+
+Imagine that during your project, have started to use a new python package not included in the original image that you already made available to other users on Docker Hub. In that case, if you have this package already installed in your image's container, you could simply use the [docker commit](https://docs.docker.com/engine/reference/commandline/commit/) command to create a new version of your image including these changes and then push it. For that you just have to run:
```
Comment (narendramukherjee) [inline content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md:194]: Didn't know this @DiSanchz , neat addition :D

**Review by narendramukherjee — APPROVED** [review]
`dockerhub.md` looks very good to me, well done!

**Comment by DiSanchz** [comment]
I need to implement a few minor modifications and after these it will be ready to merge @lachlandeer. I'll let you know once it is done!

File: content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md (hunk 2 above)
Comment (DiSanchz) [reply content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md:99]: Fixed, I've added that extra info to the paragraph!

File: content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md (hunk 1 above)
Comment (DiSanchz) [reply content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md:51]: Cool! I've included a mention to this option in the BB as a tip.

File: content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md (hunk 3 above)
Comment (DiSanchz) [reply content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md:194]: Thank you! I thought it was an interesting addition to the BB

**Comment by lachlandeer** [comment]
@shrabasteebanerjee can we merge this?

**Comment by shrabasteebanerjee** [comment]
Yes it's good to go!

**Comment by lachlandeer** [comment]
cool! then we'll merge-y  merge this bad boy in!

**Pull request merged**